### PR TITLE
Additional code cleanup and Chains enchantment bugfix

### PR DESCRIPTION
**Pull request by ghost**
Followup to [PR 128](https://github.com/chronosacaria/MCDungeonsWeapons/pull/128).

- Moved some `for` loops to stream semantics
- Simplified some implementations and removed unnecessary casts (`InventoryHelper.mcdw$hasItem`, `McdwClient.java`)
- Abstracted some code on ProjectileEffectHelper
- Fixed a possible bug on the Chain Reaction enchantment effect where the target and the attacker were swapped

### CODE REVIEW

**Comment by ghost** [comment]
The bug on the Chain Reaction enchantment seemed like an oversight, based on the argument names for the methods and the values passed in. If it was intentional, let me know and i can remove it from the PR.

File: src/main/java/chronosacaria/mcdw/client/McdwClient.java [hunk 1]
```diff
@@ -197,9 +184,9 @@ public static void registerCrossbowPredicates(McdwCrossbow crossbow) {
         });
     }
 
-    public static void registerShieldPredicates(McdwShield shield){
+    public static void registerShieldPredicates(McdwShield shield) {
         ModelPredicateProviderRegistry.register(shield, new Identifier("blocking"), (itemStack, clientWorld,
-                livingEntity, seed) -> livingEntity != null && livingEntity.isUsingItem() && livingEntity.getActiveItem()
-                == itemStack ? 1.0F : 0.0F );
+                                                                                     livingEntity, seed) -> livingEntity != null && livingEntity.isUsingItem() && livingEntity.getActiveItem()
```
Comment (Kluzzio) [inline src/main/java/chronosacaria/mcdw/client/McdwClient.java]: This indenting feels a little weird but it looks like there's no good way to do it so meh.

File: src/main/java/chronosacaria/mcdw/api/util/InventoryHelper.java [hunk 2]
```diff
@@ -15,16 +16,7 @@ public class InventoryHelper {
...
-        for (int slotID = 0; slotID < playerInventory.size(); slotID++) {
-            ItemStack currentStack = playerInventory.getStack(slotID);
-            if (currentStack.getItem() == item) {
-                count -= currentStack.getCount();
-                if (count <= 0)
-                    return true;
-            }
-        }
-        return false;
+        return mcdw$countItem(playerEntity, item) >= count;
```
Comment (Kluzzio) [inline src/main/java/chronosacaria/mcdw/api/util/InventoryHelper.java:19]: Big facepalm on my part. Good catch!

Comment (Kluzzio) [reply src/main/java/chronosacaria/mcdw/api/util/InventoryHelper.java:19]: The Chain Reaction thing is almost certainly mismatched parameters, but I'll trace it in a bit to make sure.

**Comment by Kluzzio** [comment]
The Chain Reaction thing is almost certainly mismatched parameters, but I'll trace it in a bit to make sure.

**Comment by Kluzzio** [comment]
Lol this ProjectileHelper is hurting my head to read. Too early for this

File: src/main/java/chronosacaria/mcdw/client/McdwClient.java (hunk 1 above)
Comment (ghost) [reply src/main/java/chronosacaria/mcdw/client/McdwClient.java]: That was Idea's autoformatter. Tried to remove it with the last commit but i guess it autoformatted anyways. I will add a fix

**Review by chronosacaria — APPROVED** [review]
Looks good to me.

**Pull request merged**